java/src/main/java/ai/rapids/cudf/BaseHostColumnVector.java

```diff
@@ -0,0 +1,267 @@
+/*
+ *
+ *  Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package ai.rapids.cudf;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class BaseHostColumnVector implements AutoCloseable {
+
+  private static final Logger log = LoggerFactory.getLogger(HostColumnVector.class);
+
+  protected final OffHeapState offHeap;
+  protected final DType type;
+  protected long rows;
+  protected Optional<Long> nullCount = Optional.empty();
+  protected List<BaseHostColumnVector> children = new ArrayList<>();
+
+  public BaseHostColumnVector(DType type, long rows,
+                              Optional<Long> nullCount, HostMemoryBuffer data, HostMemoryBuffer validity,
+                              HostMemoryBuffer offsets, List<BaseHostColumnVector> nestedChildren) {
+    this.offHeap = new OffHeapState(data, validity,  offsets);
+    MemoryCleaner.register(this, offHeap);
+    this.type = type;
+    this.rows = rows;
+    this.nullCount = nullCount;
+    this.children = nestedChildren;
+  }
+
+  @Override
+  public void close() {
+    for (BaseHostColumnVector child : children) {
+      if (child != null) {
+        child.close();
+      }
+    }
+    offHeap.delRef();
+    offHeap.cleanImpl(false);
+  }
+
+  HostMemoryBuffer getData() {
+    return offHeap.data;
+  }
+
+  HostMemoryBuffer getValidity() {
+    return offHeap.valid;
+  }
+
+  DType getType() {
+    return type;
+  }
+
+  long getNullCount() {
+    return nullCount.get();
+  }
+
+  List<BaseHostColumnVector> getNestedChildren() {
```

Comment:
nit: I'm not a fan of exposing `BaseHostColumnVector` as a name in the public API. It is playing a dual role as both the `Base` for `HostColumnVector` and as the children of a `HostColumnVector`.  Perhaps `HostColumnVectorCore` or `CoreHostColumnVector` might be better?

@jlowe can you think of a better name for this? or if you have a similar concern?